Assets/Countly/Plugins/CountlySDK/Services/ViewCountlyService.cs

```diff
@@ -789,59 +793,23 @@ private void UpdateGlobalViewSegmentationInternal(Dictionary<string, object> vie
         /// Do not use with <see cref="StopView"/> as it will not function correctly.
         /// Please use <see cref="StartView"/> and <see cref="StopView"/> for new implementations.
         /// </remarks>
-        /// <param name="viewName">The name of the view to open.</param>
-        /// <param name="viewSegmentation">Optional segmentation data for the view.</param>
+        /// <param name="name">The name of the view to open.</param>
+        /// <param name="segmentation">Optional segmentation data for the view.</param>
         /// <returns>A task representing the asynchronous operation.</returns>
         [Obsolete("RecordOpenViewAsync(string name, IDictionary<string, object> segmentation = null) is deprecated and will be removed in the future. Please use StartView instead.")]
         public async Task RecordOpenViewAsync(string name, IDictionary<string, object> segmentation = null)
-        {
-            lock (LockObj) {
-                Log.Info("[ViewCountlyService] RecordOpenViewAsync : name = " + name);
-
-                if (!_consentService.CheckConsentInternal(Consents.Views)) {
-                    Log.Debug("[ViewCountlyService] RecordOpenViewAsync, consent is not given, ignoring the request.");
-                    return;
-                }
-
-                if (string.IsNullOrEmpty(name)) {
-                    return;
-                }
-
-                if (name.Length > _configuration.MaxKeyLength) {
-                    Log.Verbose("[ViewCountlyService] RecordOpenViewAsync, max allowed key length is " + _configuration.GetMaxKeyLength());
-                    name = name.Substring(0, _configuration.GetMaxKeyLength());
-                }
-
-                IDictionary<string, object> openViewSegment = new Dictionary<string, object>
-                {
-                    {"name", name},
-                    {"segment", _configuration.metricHelper.OS},
-                    {"visit", 1},
-                    {"start", _isFirstView ? 1 : 0}
-                };
-
-                if (segmentation != null) {
-                    segmentation = RemoveSegmentInvalidDataTypes(segmentation);
-                    segmentation = FixSegmentKeysAndValues(segmentation);
-
-                    foreach (KeyValuePair<string, object> item in openViewSegment) {
-                        segmentation[item.Key] = item.Value;
-                    }
-                } else {
-                    segmentation = openViewSegment;
-                }
-
-                if (!_viewToLastViewStartTime.ContainsKey(name)) {
-                    _viewToLastViewStartTime.Add(name, DateTime.UtcNow);
-                }
-
-                CountlyEventModel currentView = new CountlyEventModel(viewEventKey, segmentation, 1);
-                _ = _eventService.RecordEventAsync(currentView);
-
-                _isFirstView = false;
-            }
-
-            await Task.CompletedTask;
+        {
```

Comment:
please move all checks and validations and filtering to the internal method. The method you expose to the customer should only log what provided and call the internal method